constructors/constructors.js

```diff
@@ -21,4 +21,112 @@
  *
  * This is how you would structure the game objects in an actual game
  * application in Unity or another similar framework.
- */
\ No newline at end of file
+ */
+
+class NPC {
+  constructor(props) {
+    this.state = {
+      name: props,
+      weapon: '',
+      health: '',
+      strength: '',
+      _level: 1, // <--- private? Only the DM can establish characters above level 1
```

Comment:
There is NO SUCH THING as private properties in Javascript? Why? Because there are no real classes. haha This is a common pattern though for those that want there to be privates in JS. I've seen it a lot.